### ISSUE TITLE
RFC: UX for tracing modules

### ISSUE THREAD

**Issue by cdaringe** [statoscope/statoscope#76]
Excellent tool. Thanks for your work here.

# Problem

I have a module and I have an entrypoint, but it can be challenging to trace why that module is _in_ my entrypoint's content.

# Discussion

- the webpack analyse app has a little node graph that explains the tracing, but still doesn't necessarily make it easy to visualize the import graph

What would be super cool would be a mechanism, even if only in a tree view vs a graph view to:

- select parent entrypoint/chunk/module, A
- select a child entrypoint/chunk/module, B
- render a focused/filtered linkage between A => B
    - this could more rapidly help users identify naughty imports, which is a very common use case used during bundle analysis

**Comment by smelukov** [comment]
Hi! Thanks! )
`Issuer Path` is that what you want?
Maybe entrpoint need to be added to issuer path list 🤔
<img src="https://user-images.githubusercontent.com/6654581/121780458-080e5280-cba9-11eb-9943-a02492e5e7a7.png" width="500">

**Comment by cdaringe** [comment]
I'll have to check again, but I was left to believe that Issuer Path could include _all_ paths/edges for that module, not just those narrowed to the parent.

**Comment by cdaringe** [comment]
Back! :) It would be helpful to have a mechanism that concisely explain the existence of a module of interest. Commonly, I find myself asking "why is this in my bundle" or "why is this in <some-chunk>".

If I click one of my many entrypoints, e.g. `pages/best-page`, then expand the modules tree as you have screenshotted above, the ability to filter to **only those reasons associated with the entrypoint**, vs _all entrypoints_, would be a killer start. Your idea about adding entrypoint in the list would be cool too. Perhaps a toggle in the navbar or something. In this regard, I could cut down the noise, and more quickly zero in on the relevant path.

I have an application with a module count well over 10k, & I essentially want to be able to see the edges between a module and an entrypoint:

```
PAGE_A   -- mod_1 -- mod_2 -- mod_3 \

PAGE_B   -- mod_4 /                 mod_under_investation

PAGE_C   -- mod5 ------------------/
```

The webpack analyze graph does this _decently_. Sometimes I will run a script to remove noise from my stats.json, upload it into analyze, just to see all of the edges (as shown above), while using statoscope for everything else!

**Comment by smelukov** [comment]
Wow, thanks for so detailed comment 👍
I’ve got an idea.
I’m going to implement an MVP of solution and then show you ☺️

**Comment by smelukov** [comment]
@cdaringe I've implemented an MVP to find all the paths between any module and entrypoint.
In this example, I'm trying to find all the possible ways from `node_modules/prettier/parser-babel.js` to `main` entrypoint.
Is that what you're expecting? ☺️

https://user-images.githubusercontent.com/6654581/125133254-421e3600-e10e-11eb-8821-96cda4ec68ed.mov

**Comment by cdaringe** [comment]
That looks fantastic! Your workflow looks quite easy to use: 

- Identify the module of interest,
- select the entry points tab, 
- Study the tree structure with entrypoints as the tree root and consumers of the module under investigation as leaves.

 Yes!

**Comment by smelukov** [comment]
Great! So, I'm going to make the code more production-ready and "shiny" and then publish )

**Comment by smelukov** [comment]
Published in v5.6
